controller_test.go

```diff
@@ -246,3 +246,53 @@ func TestContainLabels(t *testing.T) {
 		t.Errorf("did not expect %s to be contained in %s", notExpected, labels)
 	}
 }
+
+func TestLabelsIntersect(tt *testing.T) {
+	for _, tc := range []struct {
+		msg       string
+		a         map[string]string
+		b         map[string]string
+		intersect bool
+	}{
+		{
+			msg: "matching maps should intersect",
+			a: map[string]string{
+				"foo": "bar",
+			},
+			b: map[string]string{
+				"foo": "bar",
+			},
+			intersect: true,
+		},
+		{
+			msg: "partly matching maps should intersect",
+			a: map[string]string{
+				"foo": "bar",
+			},
+			b: map[string]string{
+				"foo": "bar",
+				"bar": "foo",
+			},
+			intersect: true,
+		},
+		{
+			msg: "maps with matching keys but different values should not inersect",
+			a: map[string]string{
+				"foo": "bar",
+				"bar": "baz",
+			},
+			b: map[string]string{
+				"foo": "bar",
+				"bar": "foo",
+			},
+			intersect: false,
+		},
```

Comment:
What about the empty label set? From this logic it seems it should intersect with another empty label set.

Reply:
Alright, the comment clears it up: empty label sets never at least one matching key/value pair.